app/models/concerns/course/levels_concern.rb

```diff
@@ -41,4 +41,30 @@ def compute_level_number(experience_points)
     level = compute_level(experience_points)
     level ? level.level_number : 0
   end
+
+  # Returns the experience point (exp) thresholds for the current and subsequent
+  # level for the given number of experience_points.
+  #
+  # [0,0] returned if levels are not present
+  # [0,x] returned if exp is 0, where x is the threshold of the first level.
+  # [x,x] returned if exp is beyond exp threshold of highest level (x)
+  #
+  # @param [Fixnum] experience_points Number of experience points
+  # @return [Array<Fixnum>] An array with two numbers, the experience_points
+  # threshold for the current and subsequent level respectively.
+  def compute_level_thresholds(experience_points)
```

Comment:
Would it be better to split this into two separate methods ?  One returns the current level's exp, the other returns the subsequent.